### PR TITLE
Feat/metadata

### PR DESCRIPTION
**Pull request by thanh-nguyen-dang**
add metadata to every record and record url.
migration script
fix unit-test
@philloooo @zflamig @giangbui

### CODE REVIEW

File: indexd/index/drivers/alchemy.py [hunk 1]
```diff
@@ -93,6 +85,40 @@ class IndexRecordUrl(Base):
...
+    url_metadata = relationship(
+        'IndexRecordUrlMetadata',
+        backref='index_record_url',
+        cascade='all, delete-orphan',
+    )
+
+
+class IndexRecordMetadata(Base):
+    '''
+        Table to track current database's schema version
```
Comment (rudyardrichter) [inline indexd/index/drivers/alchemy.py]: Indented too far; also seems like docstring should be updated?

File: indexd/index/drivers/alchemy.py [hunk 2]
```diff
@@ -93,6 +85,40 @@ class IndexRecordUrl(Base):
...
+    key = Column(String, primary_key=True)
+    did = Column(String, ForeignKey('index_record.did'), primary_key=True)
+    value = Column(String)
+
+    Index('__did_key_idx', 'did', 'key')
+
+
+class IndexRecordUrlMetadata(Base):
+    '''
+        Table to track current database's schema version
```
Comment (rudyardrichter) [inline indexd/index/drivers/alchemy.py]: Same as last comment

File: tests/alchemy_test.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
+from cdispyutils.log import get_logger
```
Comment (rudyardrichter) [inline tests/alchemy_test.py]: I would rename this file so it's clear this is for import and not to actually run any tests

File: indexd/index/drivers/alchemy.py [hunk 4]
```diff
@@ -620,6 +684,26 @@ def migrate_3(session, **kwargs):
         "CREATE INDEX {tb}__file_name_idx ON {tb} ( file_name )"
         .format(tb=IndexRecord.__tablename__))
 
+
+def migrate_4(session, **kwargs):
```
Comment (philloooo) [inline indexd/index/drivers/alchemy.py]: you don't need migration scripts for creating new table.
Base.create_all() will detect missing tables and create them

File: indexd/index/blueprint.py [hunk 5]
```diff
@@ -171,12 +176,14 @@ def post_index_record():
     size = flask.request.json['size']
     urls = flask.request.json['urls']
     hashes = flask.request.json['hashes']
-    file_name = flask.request.json.get('file_name')
+    file_name = flask.request.json.get('file_name', None)
```
Comment (rudyardrichter) [inline indexd/index/blueprint.py]: `file_name = flask.request.json.get('file_name')` equivalent

File: indexd/index/blueprint.py [hunk 6]
```diff
@@ -171,12 +176,14 @@ def post_index_record():
...
+    metadata = flask.request.json.get('metadata', None)
```
Comment (rudyardrichter) [inline indexd/index/blueprint.py]: `metadata = flask.request.json.get('metadata')` equivalent

Comment (thanh-nguyen-dang) [reply indexd/index/blueprint.py]: updated

File: indexd/index/drivers/alchemy.py (hunk 4 above)
Comment (thanh-nguyen-dang) [reply indexd/index/drivers/alchemy.py]: thanks, removed it

**Pull request merged**